zh/enable-tidb-cluster-auto-scaling.md

```diff
@@ -0,0 +1,193 @@
+      mutation:
+        pods: true
+    ```
+
+2. 安装/更新 TiDB Operator
+
+    修改完 `values.yaml` 文件中上述配置项以后进行 TiDB Operator 部署或者更新。安装与更新 Operator 请参考[在 Kubernetes 上部署 TiDB Operator](deploy-tidb-operator.md)。
+
+## 了解 TidbClusterAutoScaler
+
+我们通过 `TidbClusterAutoScaler` CR 对象来控制 TiDB 集群的弹性伸缩行为，如果你曾经使用过 [Horizontal Pod Autoscaler](https://kubernetes.io/zh/docs/tasks/run-application/horizontal-pod-autoscale/)，那么你一定会对这个概念感到非常熟悉。以下是一个 TiKV 的弹性伸缩例子。
+
+    ```yaml
+    apiVersion: pingcap.com/v1alpha1
+    kind: TidbClusterAutoScaler
+    metadata:
+      name: auto-scaling-demo
+    spec:
+      cluster:
+        name: auto-scaling-demo
+        namespace: default
+      monitor:
+        name: auto-scaling-demo
+        namespace: default
+      tikv:
+        minReplicas: 3
+        maxReplicas: 4
+        metrics:
+          - type: "Resource"
+            resource:
+              name: "cpu"
+              target:
+                type: "Utilization"
+                averageUtilization: 80
+    ```
+
+对于 TiDB 组件，你可以通过 `spec.tidb` 来进行配置，目前 TiKV 与 TiDB 的弹性伸缩 API 相同。
+
+在 `TidbClusterAutoScaler` 对象中，`cluster` 属性代表了需要被弹性调度的 TiDB 集群，通过 name 与 namespace 来标识。由于 `TidbClusterAutoScaler` 组件需要通过指标采集组件抓取相关资源使用情况，我们需要提供对应的指标采集与查询服务给 `TidbClusterAutoScaler`。`monitor` 属性则代表了与之相关连的 TidbMonitor 对象。如果你不了解 TidbMonitor，可以参考[通过 TidbMonitor 监控 TiDB 集群](monitor-using-tidbmonitor.md)。
+
+对于非 `TidbMonitor` 的外部 `Prometheus`, 你可以通过 `spec.metricsUrl` 来填写这个服务的 Host，从而指定该 TiDB 集群的监控指标采集服务。对于使用 `Helm` 部署 TiDB 集群监控的情况，可以通过以下方式来指定 `spec.metricsUrl`。
+
+    ```yaml
+    apiVersion: pingcap.com/v1alpha1
+    kind: TidbClusterAutoScaler
+    metadata:
+      name: auto-scaling-demo
+    spec:
+      cluster:
+        name: auto-scaling-demo
+        namespace: default
+      metricsUrl: "http://<release-name>-prometheus.<release-namespace>.svc:9090"
+      ......
+    ```
+
+## 快速上手
+
+我们将通过以下指令快速部署一个 3 PD、3 TiKV、2 TiDB，并带有监控与弹性伸缩能力的 TiDB 集群。
+
+```shell
+$ kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/auto-scale/tidb-cluster.yaml -n <namespace>
+tidbcluster.pingcap.com/auto-scaling-demo created
+
+$ kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/auto-scale/tidb-monitor.yaml -n <namespace>
+tidbmonitor.pingcap.com/auto-scaling-demo created
+
+$ kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/auto-scale/tidb-cluster-auto-scaler.yaml  -n <namespace>
+tidbclusterautoscaler.pingcap.com/auto-scaling-demo created
+```
+
+当 TiDB 集群创建完毕以后，你可以通过 [sysbench](https://www.percona.com/blog/tag/sysbench/) 等数据库压测工具进行压测来体验弹性伸缩功能。
+
+使用如下命令销毁环境：
+
+```shell
+kubectl delete tidbcluster auto-scaling-demo -n <namespace>
+kubectl delete tidbmonitor auto-scaling-demo -n <namespace>
+kubectl delete tidbclusterautoscaler auto-scaling-demo -n <namespace>
+```
+
+## 配置 TidbClusterAutoScaler
+
+1. 设置弹性伸缩间隔
+
+相比无状态的 Web 服务，一个分布式数据库软件对于实例的伸缩往往是非常敏感的。我们需要保证每次弹性伸缩之间存在一定的间隔，从而避免引起频繁的弹性伸缩。
```

Comment:
The sentences in the list should be indented, right? @ran-huang

Reply:
Yes, you are right.

Reply:
updated